### DISCUSSION TITLE
Advice reviving old themes.

### DISCUSSION THREAD

**Discussion by mathieujobin** [paulmcauley/klassy#8] [General]
Hi

Can you offer any advice on how one could revive old KDE themes such as the Keramik window decoration?

Also do you think Aurora is the best we can have at producing pixmap based themes or we could make it easier?

**Comment by paulmcauley** [comment]
If you already know C++ I would just fork and not bother with Aurorae as it is just too buggy, especially on HiDPI displays (at least for now until bugs are fixed).

I am not that familiar with Keramik or how it was implemented but I see this it it?
![keramik](https://user-images.githubusercontent.com/6924479/142771952-4d1b931d-3b01-4704-aa42-0f603fda279d.jpg)
It has quite an unusual highlight of the title caption, and not something I had thought about. Not trivial to implement as it also changes the window shape, but could be done.

On making customization easier, I plan to update ClassikStyles in the future to try to have the ability to load in your system's svg system icon theme for the icons. That would be allow easier and better customisation than aurorae for the vast majority of cases, and also much easier than creating icons in QPainter!

**Reply by mathieujobin** [reply]
I can't remember if this was KDE 2.x or 3.x, but Keramik used to be default theme....

Wait, are you planning ClassikStyles to be customizable to a point it could replace Aurorae?

**Reply by paulmcauley** [reply]
Now that you have jogged my memory, I do vaguely remember Keramik but never liked it and I always used the KDE1 style in the older versions of KDE, hence why I created an updated version of KDE1 in Classik 😃 (I also never liked the Oxygen/Breeze window icons and think they are unintuitive).  I probably should create a "KDE1" style fully faithful to the original as well (that would require implementing embossing effects etc. which is not my priority right now). I may also change Redmond to "Redmond 10" and have a "Redmond (pre 10)".

No, not planning to totally replace Aurorae, but taking a different approach than Aurorae's complete SVG method. I want a way to cover customization for the most common use cases, and in the medium term this would allow creating less buggy common customizations than Aurorae. I found it a pain that for any window decoration (Aurorae / C++ /QML) I then had to do duplicate work by also creating a matching icon theme to get near full system consistency, so why not cut out that extra step and have an option to just read the system icon theme directly. If Aurorae ever gets fixed to not be so buggy then of course functionality could be merged together.

**Reply by mathieujobin** [reply]
I'm afraid most of Aurorae was develop during KDE 4.x days and not much love was given to the theming capability since.
While I do prefer KDE for almost everything, I think GTK world always had an edge on theming.

**Reply by mathieujobin** [reply]
For reference, KDE 1.1 screenshot, ![KDE1.1](https://ja.wikipedia.org/wiki/K_Desktop_Environment_1#/media/%E3%83%95%E3%82%A1%E3%82%A4%E3%83%AB:KDE_1.1.jpg)

**Reply by mathieujobin** [reply]
https://invent.kde.org/plasma/kwin/-/tree/50a9e9aa2e654fc9ce53ad34c546c16f8cb9aa5f/clients

Not necessarily the last commit. But from there we could try reviving the old decorations.

I think I can find the removal commits and revert that on top of the most recent branch

Hopefully the code doesn't have too many deprecation that needs fixing

**Reply by paulmcauley** [reply]
Looking at the source code, it is significantly different and using the original KDecoration API (now KDecoration2). Probably easier to rewrite.

**Reply by mathieujobin** [reply]
Ouch, thank you for the additional information.

**Comment by paulmcauley** [comment]
I also would like to add a presets feature.

**Comment by mathieujobin** [comment]
Totally separate topic, than Keramik, but I am trying to parse E16 themes in Ruby and translate it in Aurorae/QML

https://github.com/mathieujobin/e16theme-ruby

**Reply by mathieujobin** [reply]
That idea is crazy though, and would need so much help ....

**Comment by mar1n3r0** [comment]
One more vote for Keramik. It is one of a kind and a hate/love relationship.

@paulmcauley Thumbs up for reviving the classic styles.

Here is KDE2 if you want to add it.

https://github.com/repos-holder/kdecoration2-kde2

It's kind of unfortunate how we move on and almost none of previous artwork is ported or kept. I was looking for KDE 3 wallpapers and didn't find any archive of it.

**Comment by mathieujobin** [comment]
Love your work @paulmcauley, ClassiK is looking great.
But I wonder why you associate it with KDE1 ?
They don't look so much alike to me.
![KDE1](https://cdn.arstechnica.net/wp-content/uploads/archive/images/gui/23-kde.gif)
![KDE2.1](https://nielssp.dk/assets/content/pages/guis/kde-2.1.2/welcome-screen.png)

more historical screenshots 
https://colinh-40s.blogspot.com/2010/12/evolution-of-kde-desktop.html

**Reply by paulmcauley** [reply]
The default Classik window button iconography was inspired by that of KDE1. The KDE1 window icons used an embossing/engraving effect for maximize/restore but I couldn't use that with an otherwise flat look so changed the restore icon (I also think the embossing/engraving difference between maximize and restore used in KDE1 is not so clear in meaning to the user).

My main goal is not, as some on this thread may have assumed by the name, to faithfully recreate old themes as you are trying to do. It is to create a default which I think is usably and visually better than Breeze for today. I think that these KDE1-derived icons are more intuitive than the icons introduced from Oxygen onwards. If I had wanted to faithfully recreate KDE1 exactly, I would have called it "KDE1" as it was called in KDE2/3.

A secondary goal is to allow customizability. The customizability has allowed me to easily experiment and design the way I want, and I plan more major customization options to emulate major DEs. If through these customization options it is possible to recreate old themes then that is an added benefit, but it is not actually my main goal.

**Comment by mathieujobin** [comment]
May I keep using this discussion... I realize this isn't entirely related to this project.

I'm dreaming to see older decorations ported to recent software. Furthermore, I wonder if there are enough people interested. We could build a small team. 

I found this Crystal team that is interesting.
https://github.com/shlusiak/kwin-style-crystal

the webpage shows a screenshot that seems it supported tabbed titlebar like the B2 theme.
http://www.saschahlusiak.de/crystal-kwin-decoration/

I found the commit in kwin where they remove Keramik and other themes.
https://invent.kde.org/plasma/kwin/-/commit/23d73bd6b0d77529ce9edd4c945b5249cf717d45

Apparently, they were moved to kdeartwork.
I wonder if more work was applied since... 
https://marc.info/?l=kwin&m=128100223007299&w=2

worse case; we could fetch them from kwin above's commit.
from what I understand, Aurorea based themes haven't changed much since KDE 4.3
https://invent.kde.org/plasma/kwin/-/blob/master/src/plugins/kdecorations/aurorae/theme-description

**Reply by mathieujobin** [reply]
in this commit, was removed something called decobenchmark, I wonder if that could still be useful today. it included a previewer
https://invent.kde.org/plasma/kwin/-/commit/45042010e2a02c425c10da692debce9c60538b82

**Reply by mathieujobin** [reply]
I reverted the removal on top of master in this commit https://github.com/mathieujobin/kwin/commit/945d4826c452dbd141c0e3a1c157b9017a5dcbac

nothing works likely, but if anyone else wants to try something out... 

same commit on gitlab https://invent.kde.org/mjobin/kwin/commit/945d4826c452dbd141c0e3a1c157b9017a5dcbac
